src/test/java/org/springframework/integration/kafka/rule/KafkaEmbedded.java

```diff
@@ -212,13 +251,44 @@ public void restart(final int index) throws Exception {
 		retryTemplate.execute(new RetryCallback<Void, Exception>() {
 			@Override
 			public Void doWithRetry(RetryContext context) throws Exception {
-				System.out.println("Retrying restart");
 				kafkaServers.get(index).startup();
 				return null;
 			}
 		});
 	}
 
+	public void waitUntilSynced(String topic, int brokerId) {
+		long initialTime = System.currentTimeMillis();
+		boolean canExit = false;
+		do {
+			try {
+				Thread.sleep(100);
+			}
+			catch (InterruptedException e) {
+				break;
+			}
+			canExit = true;
+			TopicMetadata topicMetadata = AdminUtils$.MODULE$.fetchTopicMetadataFromZk(topic, getZkClient());
+			if (Errors.forCode(topicMetadata.errorCode()).exception() == null) {
+				for (PartitionMetadata partitionMetadata :
+						JavaConversions.asJavaCollection(topicMetadata.partitionsMetadata())) {
+					Collection<Broker> isr = JavaConversions.asJavaCollection(partitionMetadata.isr());
```

Comment:
`isr()`?
Perfect method name! Especially when you don't have source code... :angry: